src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -0,0 +1,467 @@
+                                                                        EmberAfAttributeType attributeType, uint16_t size,
+                                                                        uint8_t * value)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    int16_t Requested;
+
+    switch (attributeId)
+    {
+    case Ids::OccupiedHeatingSetpoint: {
+        int16_t AbsMinHeatSetpointLimit; // 7C (44.5 F) is the default
+        int16_t AbsMaxHeatSetpointLimit; // 30C (86 F) is the default
+        int16_t MinHeatSetpointLimit;    // 7C (44.5 F) is the default
+        int16_t MaxHeatSetpointLimit;    // 30C (86 F) is the default
+
+        status = GetAbsMinHeatSetpointLimit(endpoint, &AbsMinHeatSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMinHeatSetpointLimit = 700;
+        }
+
+        status = GetAbsMaxHeatSetpointLimit(endpoint, &AbsMaxHeatSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMaxHeatSetpointLimit = 3000;
+        }
+
+        status = GetMinHeatSetpointLimit(endpoint, &MinHeatSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            MinHeatSetpointLimit = AbsMinHeatSetpointLimit;
+        }
+
+        status = GetMaxHeatSetpointLimit(endpoint, &MaxHeatSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            MaxHeatSetpointLimit = AbsMaxHeatSetpointLimit;
+        }
+
+        Requested = static_cast<int16_t>(chip::Encoding::LittleEndian::Get16(value));
+        if (Requested < AbsMinHeatSetpointLimit || Requested < MinHeatSetpointLimit || Requested > AbsMaxHeatSetpointLimit ||
+            Requested > MaxHeatSetpointLimit)
+            status = EMBER_ZCL_STATUS_INVALID_VALUE;
+
+        break;
+    }
+
+    case Ids::OccupiedCoolingSetpoint: {
+        int16_t AbsMinCoolSetpointLimit; // 16C (61 F) is the default
+        int16_t AbsMaxCoolSetpointLimit; // 32C (90 F) is the default
+        int16_t MinCoolSetpointLimit;    // 16C (61 F) is the default
+        int16_t MaxCoolSetpointLimit;    // 32C (90 F) is the default
+
+        status = GetAbsMinCoolSetpointLimit(endpoint, &AbsMinCoolSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMinCoolSetpointLimit = 1600;
+        }
+
+        status = GetAbsMaxCoolSetpointLimit(endpoint, &AbsMaxCoolSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            AbsMaxCoolSetpointLimit = 3200;
+        }
+
+        status = GetMinCoolSetpointLimit(endpoint, &MinCoolSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
+        {
+            MinCoolSetpointLimit = AbsMinCoolSetpointLimit;
+        }
+
+        status = GetMaxCoolSetpointLimit(endpoint, &MaxCoolSetpointLimit);
+        if (status != EMBER_ZCL_STATUS_SUCCESS)
```

Comment:
same as above here and on line 158 below: the last `staus = ` will bleed through the return, even though the default value is set.

Reply:
Added else case so switch case will always return success or invalid value